packages/@azure/servicebus/data-plane/lib/receiver.ts

```diff
@@ -240,3 +240,219 @@ export class Receiver {
+   * @returns Promise<ServiceBusMessage[]>
+   * - Returns a list of messages identified by the given sequenceNumbers.
+   * - Returns an empty list if no messages are found.
+   * - Throws an error if the messages have not been deferred.
+   */
+
+  async receiveDeferredMessages(sequenceNumbers: Long[]): Promise<ServiceBusMessage[]> {
+    if (this._receiveMode !== ReceiveMode.peekLock) {
+      throw new Error("The operation is only supported in 'PeekLock' receive mode.");
+    }
+
+    return this._context.managementClient!.receiveDeferredMessages(
+      sequenceNumbers,
+
+      this._receiveMode,
+
+      this.sessionId
+    );
+  }
+
+  /**
+   * Returns a batch of messages based on given count and timeout over an AMQP receiver link
+   * from a Queue/Subscription.
+   *
+   * @param maxMessageCount      The maximum number of messages to receive from Queue/Subscription.
+   * @param maxWaitTimeInSeconds The maximum wait time in seconds for which the Receiver
+   * should wait to receive the first message. If no message is received by this time,
+   * the returned promise gets resolved to an empty array.
+   * - **Default**: `60` seconds.
+   * @returns Promise<ServiceBusMessage[]> A promise that resolves with an array of Message objects.
+   */
+
+  async receiveBatch(
+    maxMessageCount: number,
+    maxWaitTimeInSeconds?: number
+  ): Promise<ServiceBusMessage[]> {
+    try {
+      return await this._sessionReceiver.receiveBatch(maxMessageCount, maxWaitTimeInSeconds);
+    } catch (err) {
+      log.error(
+        "[%s] Receiver '%s', an error occurred while receiving %d messages for %d " +
+          "max time:\n %O",
+        this._context.namespace.connectionId,
+        this._sessionReceiver.name,
+        maxMessageCount,
+        maxWaitTimeInSeconds,
+        err
+      );
+      throw err;
+    }
+  }
+
+  /**
+   * Registers handlers to deal with the incoming stream of messages over an AMQP receiver link
+   * from a Queue/Subscription.
+   * To stop receiving messages, call `close()` on the SessionReceiver or set the property
+   * `newMessageWaitTimeoutInSeconds` in the options to provide a timeout.
+   *
+   * @param onMessage - Handler for processing each incoming message.
+   * @param onError - Handler for any error that occurs while receiving or processing messages.
+   * @param options - Options to control whether messages should be automatically completed. You can
+   * also provide a timeout in seconds to denote the amount of time to wait for a new message
+   * before closing the receiver.
+   *
+   * @returns void
+   */
+  receive(onMessage: OnMessage, onError: OnError, options?: SessionMessageHandlerOptions): void {
+    try {
+      return this._sessionReceiver.receive(onMessage, onError, options);
+    } catch (err) {
+      const msg =
+        `MessageSession with sessionId '${this.sessionId}' and name '${
+          this._sessionReceiver.name
+        }' ` + `has either not been created or is not open.`;
```

Comment:
This catch block is for any error inside `this._sessionReceiver.receive`
The error message is very specific to receiver not being open or created.
We should mimic what is happening in the `recieve` of the normal `receiver`